File: aforclinic-api-main/aforclinic-api-main/src/main/java/com/xenialsoft/api/core/mainbanner/model/MainBanner.java
package com.xenialsoft.api.core.mainbanner.model;

import com.xenialsoft.api.common.Auditable;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class MainBanner extends Auditable {

	private String id;

	private String title;
	
	private String imageId;

	private String linkUri;

	private int displayOrder;

	private MainBannerStatus status;

}

File: aforclinic-web-main/aforclinic-web-main/shared/types/MainBanner.ts
export type MainBannerStatus = 'ACTIVE' | 'DELETED';

export interface MainBanner {
  /**
   * 순번
   */
  sequence?: number;
  /**
   * 메인화면 배너 식별자
   */
  id: string;
  /**
   * 제목
   */
  title: string;
  /**
   * 이미지 식별자
   */
  imageId: string;
  /**
   * 링크
   */
  linkUri: string;
  /**
   * 화면 표시 순서
   */
  displayOrder?: number;
  /**
   * 상태
   */
  status?: MainBannerStatus;
  /**
   * 등록일시
   */
  createdAt?: string;
  /**
   * 등록자
   */
  createdBy?: string;
  /**
   * 수정일시
   */
  modifiedAt?: string;
  /**
   * 수정자
   */
  modifiedBy?: string;
};

export type CreateMainBanner = Pick<MainBanner, 'title' | `imageId` | `linkUri` | 'displayOrder'>;
export type UpdateMainBanner = Pick<MainBanner, 'title' | 'imageId' | `linkUri` | 'displayOrder' | 'status'>;
export type UpdateMainBannerStatus = Pick<MainBanner, 'status'>;
export type ReorderMainBanner = Pick<MainBanner, 'id' | 'displayOrder'>;
